lib/Driver/DarwinToolChains.cpp

```diff
@@ -854,6 +854,12 @@ toolchains::Darwin::validateArguments(DiagnosticEngine &diags,
     diags.diagnose(SourceLoc(), diag::error_darwin_static_stdlib_not_supported);
   }
 
+  // Validating darwin deprecated -link-objc-runtime.
+  if (args.hasArg(options::OPT_link_objc_runtime,
+		  options::OPT_no_link_objc_runtime)) {
+    diags.diagnose(SourceLoc(), diag::warn_darwin_link_objc_deprecated);
```

Comment:
@tshortli Like this?   (I'm not sure this is the right place for that diagnostic; if you think there's a better place, please let me know.)

Reply:
Yes that looks reasonable to me! There may be some tests that need to be updated to expect this diagnostic now.